### PR TITLE
handle couch roles with missing attrs

### PR DESCRIPTION
**Pull request by snopoke**
## Summary
Fix key errors in `populate_user_role` command 

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
NA

### Safety story
update to management command

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
test failure looks real
```
ERROR: corehq.apps.users.tests.test_migrate_roles_to_sql:UserRoleCouchToSqlTests.test_diff_assignable_by_couch_None
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/users/tests/test_migrate_roles_to_sql.py", line 144, in test_diff_assignable_by_couch_None
    self.assertListEqual(Command.get_filtered_diffs(couch.to_json(), sql), [
  File "/mnt/commcare-hq-ro/corehq/apps/cleanup/management/commands/populate_sql_model_from_couch_model.py", line 66, in get_filtered_diffs
    diffs = cls.diff_couch_and_sql(couch, sql)
  File "/mnt/commcare-hq-ro/corehq/apps/users/management/commands/populate_user_role.py", line 55, in diff_couch_and_sql
    sorted(couch_assignable_by),
TypeError: 'NoneType' object is not iterable
```

**Review by millerdev — COMMENTED** [review]
How hard would it be to write a test for the bug this is fixing?

File: corehq/apps/users/management/commands/populate_user_role.py [hunk 1]
```diff
@@ -48,11 +48,11 @@ def diff_couch_and_sql(cls, couch, sql):
                 name_prefix=f"permissions.{name}"
             ))
 
-        couch_assignable_by = couch["assignable_by"]
+        couch_assignable_by = couch.get("assignable_by", [])
```
Comment (snopoke) [inline corehq/apps/users/management/commands/populate_user_role.py:51]:
```suggestion
        couch_assignable_by = couch.get("assignable_by")
```

File: corehq/apps/users/management/commands/populate_user_role.py [hunk 2]
```diff
@@ -48,11 +48,11 @@ def diff_couch_and_sql(cls, couch, sql):
...
         sql_assignable_by = list(sql.roleassignableby_set.values_list('assignable_by_role__couch_id', flat=True))
         diffs.extend(cls.diff_lists(
             "assignable_by",
-            sorted(couch_assignable_by) if couch_assignable_by else [],
+            sorted(couch_assignable_by),
```
Comment (snopoke) [inline corehq/apps/users/management/commands/populate_user_role.py:55]:
```suggestion
            sorted(couch_assignable_by) if couch_assignable_by else [],
```

**Comment by snopoke** [comment]
> How hard would it be to write a test for the bug this is fixing?

Not very hard but there's not a lot of value since this is a once off mgmt command and I've already used this code on a few envs.

**Review by millerdev — APPROVED** [review]

**Pull request merged**